packages/preview-middleware/src/base/flp.ts

```diff
@@ -227,9 +227,13 @@ export class FlpSandbox {
                 previewUrl = `${previewUrl}.inner.html`;
                 editor.pluginScript ??= 'open/ux/preview/client/cpe/init';
                 this.router.get(editor.path, (_req: Request, res: Response) => {
+                    let templatePreviewUrl = `${previewUrl}?sap-ui-xx-viewCache=false&fiori-tools-rta-mode=forAdaptation&sap-ui-rta-skip-flex-validation=true&sap-ui-xx-condense-changes=true#${this.config.intent.object}-${this.config.intent.action}`;
+                    if (rta.options?.scenario === 'ADAPTATION_PROJECT') {
+                        templatePreviewUrl = templatePreviewUrl.replace('?', `?sap-ui-layer=${rta.layer}&`);
```

Comment:
I don't get it. Why do you need to put the layer as an URL parameter? Please elaborate.

Reply:
Clarified.